corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/api.js

```diff
@@ -29,7 +29,13 @@ hqDefine("cloudcare/js/formplayer/menus/api", function () {
                 defer = $.Deferred(),
                 options,
                 menus;
-
+                sessionStorage.formplayerQueryInProgress = true;
+                setTimeout(() => {
+                    if (!$('.duration-to-show-loading-met').length) {
+                        const newdiv1 = $("<div class='duration-to-show-loading-met' style='display: none;'></div>");
```

Comment:
From what i'm understanding, adding these elements are only so that the listener in `loadingDurationListener` has something to observe? I can't think of a better way, but I feel like there would be a less complex way. 

I can't tell currently how the progress bar is triggered to be shown, but I imagine it's done via a call to `showLoading`. Could the setTimeout be done directly in there and store a reference to it? and then wherever the loading bar should be cleared, we also clear the setTimeout?

Reply:
Yeah, this element is added specifically as a flag for the listener. I also felt like there should be a less complex way of setting this up, but this configuration is the one that has been stable/dependable. I've tried using `setTimeout` in `showProminentLoading` method and just checking `sessionStorage.formplayerQueryInProgress`, but that introduces a bug where in some cases you get a progress bar that doesn't clear. There may still be a way to make it work without the listener, but I haven't found it.
The `setTimeout` doesn't need clearing like `setInterval` it just runs the once and that's it.

Reply:
@Jtang-1 I'm revisiting this after fixing a bug and I think it is possible/preferable to handle the `setTimeout` and `sessionStorage.formplayerQueryInProgress` in utils. Changes coming shortly.

Reply:
Nice, that looks way cleaner